indico/modules/events/abstracts/controllers/common.py

```diff
@@ -42,7 +43,12 @@ def _process(self):
         return self._render_template(**self.list_generator.get_list_kwargs())
 
     def _render_template(self, **kwargs):
-        return self.view_class.render_template(self.template, self.event, **kwargs)
+        abstract_files_count = (AbstractFile.query
+                                .filter(AbstractFile.abstract.has(event=self.event, is_deleted=False))
+                                .count())
+        can_download_attachments = self.event.cfa.allow_attachments or abstract_files_count
```

Comment:
You can optimize this to only query if attachments are disabled for the event. Also, `has_rows()` is a bit more efficient since you don't need the actual count, just whether there are any rows or not.

```
can_download_attachments = self.event.cfa.allow_attachments
if not can_download_attachments:
    can_download_attachments = (AbstractFile.query
                                .filter(AbstractFile.abstract.has(event=self.event, is_deleted=False))
                                .has_rows())
```